app/scripts/controllers/serviceInstance.js

```diff
@@ -42,28 +65,76 @@ angular.module('openshiftConsole')
       });
     };
 
-    var serviceClassPromise;
+    var updateParameterData = function() {
+      if (!$scope.serviceInstance || !$scope.parameterSchema) {
+        return;
+      }
+
+      DataService.unwatchAll(secretWatchers);
+      secretWatchers = [];
+
+      $scope.parameterData = {};
+      _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function(key) {
+        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      });
+
+      $scope.parameterData = angular.extend($scope.parameterData, _.get($scope.serviceInstance, 'spec.parameters', {}));
+
+      if (AuthorizationService.canI('secrets', 'get', $scope.projectName)) {
```

Comment:
This means that you will not be able to see that there's a parameter at all when it comes from a secret and you can't get the secret, correct?

We could possibly fill in the redacted values using `status.externalProperties.parameters` when users can't list secrets (without letting them reveal). I'm OK handling this as a follow on, however.

Reply:
@spadgett  What should we do when the user does reveal and we have redacted values for some of the parameters? Continue to show ***** for those parameters?

Reply:
Maybe remove the "reveal" action?

Reply:
I think we're close enough here it should be a follow-on anyway.